pennylane/math/interface_utils.py

```diff
@@ -46,6 +46,14 @@ def get_like(self):
         }
         return mapping[self]
 
+    def __eq__(self, interface):
+        if isinstance(interface, str):
+            raise TypeError("Cannot compare Interface with str")
```

Comment:
```suggestion
            return self.value == interface
```
Then we don't even need to ban the comparison with str and the initial bug should be also fixed

Reply:
I think we should ban it though. Internally, we should be using the "canonical" interface a.k.a the Enum to standardize everything. If we have to rely on strings it could get ugly.

Reply:
The only time we should be dealing with strings is from a user, then immediately convert it to a Enum object.